packages/itwin/grouping-mapping-widget/src/test/QueryBuilder.test.ts

```diff
@@ -2,178 +2,67 @@
+  const queryBuildertestData: QueryBuilderTestData = testData;
+  queryBuildertestData.testCases.forEach((testCase) => {
+    it(testCase.name, async () => executeTest(
+      sut,
+      testCase.expectedResult,
+      testCase.operations,
+    ));
+  });
 
-      testBuildQueryString(query, "SELECT BaseClassName.ECInstanceId FROM BaseClassName JOIN JoinClassName JOIN bis.Category ON bis.Category.ECInstanceId = bis.GeometricElement3d.category.id AND ((bis.Category.CodeValue='propValue1') OR (bis.Category.UserLabel='propValue1'))");
-    });
+  const mockPropertiesField = (propertyRecord: PropertyRecord, propertiesField: PropertiesField) => {
+    const propertiesFieldMock: StubbedType<PropertiesField> = MockFactory.create(PropertiesField);
 
-    it("should return query string with where clause, when base class has properties", () => {
-      const baseClass: QueryClass = {
-        className: "BaseClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [
-          {
-            name: "propName1",
-            needsQuote: false,
-            isCategory: false,
-            value: "propValue1",
-          },
-          {
-            name: "propName2",
-            needsQuote: false,
-            isCategory: false,
-            value: "propValue2",
-          },
-        ],
-      };
+    MockFactory.stubProperty(propertiesFieldMock, "parent", () => propertiesField.parent);
+    MockFactory.stubProperty(propertiesFieldMock, "properties", () => propertiesField.properties);
+    MockFactory.stubProperty(propertiesFieldMock, "type", () => propertiesField.type);
 
-      const joinClass: QueryClass = {
-        className: "JoinClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [],
-      };
+    dataProvider.getFieldByPropertyRecord.withArgs(propertyRecord).returns(Promise.resolve(propertiesFieldMock));
+  };
 
-      const query: Query = {
-        unions: [
-          {classes: [baseClass, joinClass]},
-        ],
-      };
-      testBuildQueryString(query, "SELECT BaseClassName.ECInstanceId FROM BaseClassName JOIN JoinClassName WHERE BaseClassName.propName1=propValue1 AND BaseClassName.propName2=propValue2");
-    });
+  const executeTest = async (queryBuilder: QueryBuilder, expectedResult: string, operations: any[]) => {
+    for (const op of operations) {
+      const prop: PropertyRecord = new PropertyRecord(op.propertyRecord.value as PropertyValue, op.propertyRecord.property as PropertyDescription);
 
-    it("should return query string with UNION, when there are multiple unions", () => {
-      const queryClass: QueryClass = {
-        className: "ClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [
-          {
-            name: "propName1",
-            needsQuote: false,
-            isCategory: false,
-            value: "propValue1",
-          },
-        ],
-      };
+      mockPropertiesField(prop, op.propertiesField);
```

Comment:
`mockPropertiesField` should be `createPropertyRecord` and it should return the `PropertyRecord`.

Reply:
Changed

Reply:
resolved